lib/__tests__/path.test.ts

```diff
@@ -1,139 +1,241 @@
+    });
+  });
+
+  describe('getExt()', () => {
+    test('returns lowercase extension without dot', () => {
+      expect(getExt('file.TXT')).toBe('txt');
+    });
+
+    test('returns empty string for no extension', () => {
+      expect(getExt('file')).toBe('');
+    });
+
+    test('handles non-string input', () => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      expect(getExt(null as '')).toBe('');
+    });
+  });
+
+  describe('getAllowedExtensions()', () => {
+    test('returns default allowed extensions', () => {
+      const result = getAllowedExtensions();
+      expect(result).toBeInstanceOf(Set);
+      expect(result).toEqual(new Set(ALLOWED_EXTENSIONS));
+    });
+
+    test('includes additional extensions', () => {
+      const result = getAllowedExtensions(['custom']);
+      expect(result.has('custom')).toBe(true);
+    });
+  });
+
+  describe('isAllowedExtension()', () => {
+    test('returns true for allowed extension', () => {
+      expect(isAllowedExtension('file.txt')).toBe(true);
+    });
+
+    test('returns false for disallowed extension', () => {
+      expect(isAllowedExtension('file.exe')).toBe(false);
+    });
+
+    test('allows custom extensions', () => {
+      expect(isAllowedExtension('file.custom', ['custom'])).toBe(true);
+    });
+  });
+
+  describe('sanitizeFileName()', () => {
+    test('replaces invalid characters', () => {
+      expect(sanitizeFileName('file:name?.txt')).toBe('file-name-.txt');
+    });
+
+    test('handles reserved names', () => {
+      expect(sanitizeFileName('CON')).toBe('-CON');
+    });
+
+    test('removes trailing periods and spaces', () => {
+      expect(sanitizeFileName('file.txt. ')).toBe('file.txt');
+    });
+  });
+
+  describe('isValidPath()', () => {
+    test('returns true for valid path', () => {
+      expect(isValidPath('/valid/path/file.txt')).toBe(true);
+    });
+
+    test('returns false for path with invalid characters', () => {
+      expect(isValidPath('/invalid/path/file?.txt')).toBe(false);
+    });
+
+    test('returns false for reserved names', () => {
+      expect(isValidPath('/some/path/CON')).toBe(false);
+    });
+  });
+
+  describe('untildify()', () => {
+    const originalHomedir = os.homedir;
+
+    beforeEach(() => {
+      (os.homedir as jest.Mock) = jest.fn().mockReturnValue('/home/user');
+    });
+
+    afterEach(() => {
+      os.homedir = originalHomedir;
+    });
+
+    test('replaces tilde with home directory', () => {
+      expect(untildify('~/documents/file.txt')).toBe(
+        '/home/user/documents/file.txt'
+      );
+    });
+
+    test('does not modify paths without tilde', () => {
+      expect(untildify('/absolute/path/file.txt')).toBe(
+        '/absolute/path/file.txt'
+      );
+    });
+
+    test('throws TypeError for non-string input', () => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
This is just a FYI, you don't need to change it.  But using `// @ts-expect-error The reason for the error` shouldn't trigger `eslint` so you can get away with just the one comment

Reply:
Ooh, nice, thanks for the tip!